### PR TITLE
Add async utils from antsibull-core

### PR DESCRIPTION
**Pull request by felixfontein**
Disadvantage of having this here: antsibull-changelog so far doesn't depend on aiofiles.

### CODE REVIEW

**Comment by gotmax23** [comment]
One idea is to make this a dependency-less package and have one extra per module. Each module/package can have something like

``` python
try:
    import module
    import other_module
except ImportError as exc:
    raise ImportError(f"{__package__} requires the hashing/io/whatever extra. Please install antsibull-fileutils[EXTRA_NAME]") from exc
```

On the other hand, aiofiles has no dependencies itself, and it doesn't compile any C extensions or anything, so perhaps not a big deal to add as a hard dependency.

**Comment by felixfontein** [comment]
After thinking some more about this, I'm for simply merging this and thus adding `aiofiles` as a dependency.

**Comment by felixfontein** [comment]
I've adjusted https://github.com/ansible-community/antsibull-core/pull/166, https://github.com/ansible-community/antsibull/pull/619, and https://github.com/ansible-community/antsibull-docs/pull/322 to use the code from this PR.

**Comment by gotmax23** [comment]
> After thinking some more about this, I'm for simply merging this and thus adding `aiofiles` as a dependency.

Sounds good to me!

File: pyproject.toml [hunk 1]
```diff
@@ -64,12 +65,15 @@ formatters = [
     "isort",
 ]
 test = [
+    "asynctest",
```
Comment (gotmax23) [inline pyproject.toml]: I don't think we use this.

```suggestion
```

**Comment by felixfontein** [comment]
@gotmax23 thanks for reviewing this!

**Pull request merged**